jupyterhub/templates/image-puller/_helper.yaml

```diff
@@ -1,28 +1,40 @@
-{{ define "jupyterhub.prePuller.daemonset" }}
-# template that returns an prePuller daemonset
+{{ define "jupyterhub.imagePuller.daemonset" }}
+# template that returns a image-puller daemonset, used for
+# - pre-helm-upgrade image pulling (pulls images before helm upgrades, temporary)
+# - continuous image pulling (pulls images for cluster autoscalers, persistent)
 apiVersion: extensions/v1beta1
 kind: DaemonSet
 metadata:
   name: {{ .name }}
   labels:
     app: jupyterhub
-    component: image-puller
+    component: {{ .name }}
     release: {{ .top.Release.Name }}
     heritage: {{ .top.Release.Service }}
+    {{ if .hook }}
+    hook-type: temporary
```

Comment:
Labels that we make up should generally be namespaced. So perhaps something like `hub.jupyter.org/deletable: "true"`?

Reply:
Excellent done!